### PR TITLE
Fix new lines with TextMsg and SayText (side-effect of #763)

### LINKED ISSUE #762: Update AMXX with latest CS 1.6 official release with % display changes.
This got fixed on Arkshine github 3 months ago but did not got released on AMXX officially and should be addressed now that CS got the update officially.

https://github.com/Arkshine/amxmodx/commit/d430021d211b9b91c06944c62f647c3b9486f62d

Issue: https://github.com/ValveSoftware/halflife/issues/2611

### LINKED ISSUE #774: No newline in console!
`----- List of loaded plugins ----- Names of version Author file Compost Anti Fullupdate S 2.0 Pike fullupdatefix.a running ReAimDetector API 0.2.2 ReHLDS Team reaimdetector.a running AMXBans: Core Gm 1.6 Larte Team amxbans_core.am running FreshBans 1.3.6 kanagava fresh_bans_136. running Rechecker Log 0.1 F @ nt0M rc_log.amxx running Teleport Menu 1.10.0.538 AMXX Dev Team telemenu.amxx running Mode 2x2 1.9 s1lent mode.amxx running Gag Me 2.3.4 Radius gag_me.amxx running Fix AutoBuy Bug 1.1 PomanoB & UFPS.T autobuyfix11.am running Admin Commands 1.10.0.538 AMXX Dev Team admincmd.amxx running Only 10 plugins, 10 of them running ----- List of available commands from 1 to 10. Total teams - 62 --------- - Write 'amx_plugins 11 "to continue ----- #

`

but need

-```
---- List of loaded plugins ----- 
Names of version Author file Compost 
Anti Fullupdate S 2.0 Pike fullupdatefix.a running
 ReAimDetector API 0.2.2 ReHLDS Team reaimdetector.a running
 AMXBans: Core Gm 1.6 Larte Team amxbans_core.am running
 FreshBans 1.3.6 kanagava fresh_bans_136. running
 Rechecker Log 0.1 F @ nt0M rc_log.amxx running
 Teleport Menu 1.10.0.538 AMXX Dev Team telemenu.amxx running
 Mode 2x2 1.9 s1lent mode.amxx running
 Gag Me 2.3.4 Radius gag_me.amxx running
 Fix AutoBuy Bug 1.1 PomanoB & UFPS.T autobuyfix11.am running
 Admin Commands 1.10.0.538 AMXX Dev Team admincmd.amxx running
 Only 10 plugins, 10 of them running
 ----- List of available commands from 1 to 10. Total teams - 62
 --------- - Write 'amx_plugins 11 "to continue ----- #

```

### LINKED ISSUE #776: [DoD] client_print's displaying as "%s" in chat
Happening even on a completely fresh stock hlds + metamod + stock amxx. 'timeleft' 'nextmap' etc all just show "%s" output in the chat.

Updated my hlds yesterday, build now showing as Jul 24 2019 (8308). Then I updated my amxx to 5386 and this started happening. Reverted back to 5378 and it works fine. Issue starts after the change in 5380.

Day of Defeat
Metamod 1.21p38
Windows Server 2008 R2

Screenshot: https://imgur.com/g1mSagN

### PR DESCRIPTION
**Pull request by Arkshine**
Related to #763.

* Use format string only for official mods. Currently, there is an exception for DoD, but it should be supported in the next game release. Ref: https://github.com/ValveSoftware/halflife/issues/2744
* Use double new lines for `TextMsg` with the following exceptions (ref: https://github.com/ValveSoftware/halflife/issues/2611#issuecomment-546454868):
  * Not an official game or DoD
  * CS with `print_center` or `print_chat`
* CS and `SayText` only: Removed new line and adjusted max length used to truncate string to reflect the format string change.

@WPMGPRoSToTeMa Will test again, but just in case, did I miss something?
I know we can improve and reduce code, can be done another time.

Fixes #762 
Fixes #774 
Fixed #776

### CODE REVIEW

File: amxmodx/amxmodx.cpp [hunk 1]
```diff
@@ -194,10 +194,10 @@ static cell AMX_NATIVE_CALL console_print(AMX *amx, cell *params) /* 2 param */
 
 	int len;
 	char* message = format_amxstring(amx, params, 2, len);
-
+	
 	if (index < 1 || index > gpGlobals->maxClients)	// Server console
 	{
-		if (len > 254)
+		if (len > 254) // Server console truncates after byte 255. (254 + \n = 255)
```
Comment (WPMGPRoSToTeMa) [inline amxmodx/amxmodx.cpp]: I just tried `EngFunc_ServerPrint` and it doesn't truncate after byte 255.

File: amxmodx/amxmodx.cpp [hunk 2]
```diff
@@ -216,18 +216,26 @@ static cell AMX_NATIVE_CALL console_print(AMX *amx, cell *params) /* 2 param */
...
 				}
 			}
-			message[len++] = '\n';      // Client expects newline from the server
+			message[len++] = '\n';
+
+			const auto canUseFormatString = g_official_mod && !g_bmod_dod; // Temporary exclusion for DoD until officially supported
+
+			if (canUseFormatString)
+			{
+				message[len++] = '\n';    //  Double newline is required when pre-formatted string in TextMSg is passed as argument.
```
Comment (WPMGPRoSToTeMa) [inline amxmodx/amxmodx.cpp]: nit: `TestMSg` -> `TextMsg`.

File: amxmodx/amxmodx.cpp [hunk 3]
```diff
@@ -252,16 +262,26 @@ static cell AMX_NATIVE_CALL client_print(AMX *amx, cell *params) /* 3 param */
...
 					{
 						len -= UTIL_CheckValidChar(msg + len - 1); // Don't truncate a multi-byte character
 					}
 				}
-				msg[len++] = '\n';	// Client expects newline from the server
+				msg[len++] = '\n';
+				
+				if (canUseFormatString && (!g_bmod_cstrike || (params[2] == HUD_PRINTNOTIFY || params[2] == HUD_PRINTCONSOLE)))
+				{
+					msg[len++] = '\n';  // Double newline is required when pre-formatted string in TextMSg is passed as argument.
```
Comment (WPMGPRoSToTeMa) [inline amxmodx/amxmodx.cpp]: nit: `TextMSg` -> `TextMsg`.

File: amxmodx/amxmodx.cpp [hunk 4]
```diff
@@ -252,16 +262,26 @@ static cell AMX_NATIVE_CALL client_print(AMX *amx, cell *params) /* 3 param */
...
+					len = bytesLimit;
 					if ((msg[len - 1] & 1 << 7))
 					{
 						len -= UTIL_CheckValidChar(msg + len - 1); // Don't truncate a multi-byte character
 					}
 				}
-				msg[len++] = '\n';	// Client expects newline from the server
+				msg[len++] = '\n';
+				
+				if (canUseFormatString && (!g_bmod_cstrike || (params[2] == HUD_PRINTNOTIFY || params[2] == HUD_PRINTCONSOLE)))
```
Comment (WPMGPRoSToTeMa) [inline amxmodx/amxmodx.cpp]: This condition looks too complicated, can we split it up to make it more readable?

File: amxmodx/amxmodx.cpp (hunk 1 above)
Comment (Arkshine) [reply amxmodx/amxmodx.cpp]: I just paste the old comment. Just tested fastly and the limit should be 1024. Will make another PR for that, it's untouched and unrelated to new lines.

**Comment by Arkshine** [comment]
Alright, I've tested on CS, DoD, DMC, OpFor, TFC, Ricochet, and Valve.
This should hopefully restore the original behavior.

**Pull request merged**